### PR TITLE
[Doc] add spark connector

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: docs/loading/Spark-connector-starrocks.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+You can obtain the Spark connector JAR file in the following ways:
+
+- Directly download the compiled Spark Connector JAR file.
+- Add the Spark connector as a dependency in your Maven project and then download the JAR file.
+- Compile the source code of the Spark Connector into a JAR file by yourself.
+
+The naming format of the Spark connector JAR file is `starrocks-spark-connector-${spark_version}_${scala_version}-${connector_version}.jar`.
+
+For example, you install Spark 3.2 and Scala 2.12 in your environment, and you want to use Spark connector 1.1.0. You can use `starrocks-spark-connector-3.2_2.12-1.1.0.jar`.
```
Comment (amber-create) [inline docs/loading/Spark-connector-starrocks.md:21]: unnatual flow.
suggest changing the sentence to "For example, if you install Spark 3.2 and Scala 2.12 in your environment and you want to use Spark connector 1.1.0, you can use `starrocks-spark-connector-3.2_2.12-1.1.0.jar`." OR "For example, you install Spark 3.2 and Scala 2.12 in your environment, and you want to use Spark connector 1.1.0. In this situation, you can use `starrocks-spark-connector-3.2_2.12-1.1.0.jar`."

**Review by amber-create — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by EsoragotoSpirit — APPROVED** [review]

**Comment by hellolilyliuyi** [comment]
@mergify backport branch-3.1 branch-3.0 branch-2.5

**Pull request merged**